sdk/src/main/scala/io/horizen/account/state/ForgerStakeV2MsgProcessor.scala

```diff
@@ -81,21 +87,75 @@ case class ForgerStakeV2MsgProcessor(networkParams: NetworkParams) extends Nativ
     Array.emptyByteArray
   }
 
-}
+  def doActivateCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+
+    //Check is well formed
+    requireIsNotPayable(invocation)
+    checkInputDoesntContainParams(invocation)
+
+    //Check it cannot called twice
+    if (StakeStorage.isActive(view)) {
+      val msgStr = s"Forger stake V2 already activated"
+      log.debug(msgStr)
+      throw new ExecutionRevertedException(msgStr)
+    }
+
+    val intrinsicGas = invocation.gasPool.getUsedGas
+
+    //Get all existing stakes from old native contract
+    val result = context.execute(invocation.call(FORGER_STAKE_SMART_CONTRACT_ADDRESS, BigInteger.ZERO,
```

Comment:
let's double check that we don't go outofgas even with 5000 stakes (the maximum number we can reach before the HF)
maybe is better to use the paginated one?

Reply:
I have modified the test so that there are 5000 stakes to migrate and it actually runs out of gas. So I have modified the disable method so it also returns the existing stakes without using gas. I have preferred this way than using the paginated methods, just in case we decided to also delete the old stakes from the old native contract.